core/primitives/src/block.rs

```diff
@@ -86,55 +86,86 @@ pub enum Block {
     BlockV4(Arc<BlockV4>),
 }
 
+#[cfg(feature = "solomon")]
+type ShardChunkReedSolomon = reed_solomon_erasure::galois_8::ReedSolomon;
+
 #[cfg(feature = "solomon")]
 pub fn genesis_chunks(
     state_roots: Vec<crate::types::StateRoot>,
+    congestion_infos: Vec<Option<CongestionInfo>>,
     shard_ids: &[crate::types::ShardId],
     initial_gas_limit: Gas,
     genesis_height: BlockHeight,
     genesis_protocol_version: ProtocolVersion,
 ) -> Vec<crate::sharding::ShardChunk> {
-    let rs = reed_solomon_erasure::galois_8::ReedSolomon::new(1, 2).unwrap();
+    let rs = ShardChunkReedSolomon::new(1, 2).unwrap();
     let state_roots = if state_roots.len() == shard_ids.len() {
         state_roots
     } else {
         assert_eq!(state_roots.len(), 1);
         std::iter::repeat(state_roots[0]).take(shard_ids.len()).collect()
     };
 
-    let congestion_info = near_primitives_core::version::ProtocolFeature::CongestionControl
-        .enabled(genesis_protocol_version)
-        .then_some(crate::congestion_info::CongestionInfo::default());
-
-    shard_ids
-        .into_iter()
-        .zip(state_roots)
-        .map(|(&shard_id, state_root)| {
-            let (encoded_chunk, _) = crate::sharding::EncodedShardChunk::new(
-                CryptoHash::default(),
-                state_root,
-                CryptoHash::default(),
-                genesis_height,
-                shard_id,
-                &rs,
-                0,
-                initial_gas_limit,
-                0,
-                CryptoHash::default(),
-                vec![],
-                vec![],
-                &[],
-                CryptoHash::default(),
-                congestion_info,
-                &crate::validator_signer::EmptyValidatorSigner::default().into(),
-                genesis_protocol_version,
-            )
-            .expect("Failed to decode genesis chunk");
-            let mut chunk = encoded_chunk.decode_chunk(1).expect("Failed to decode genesis chunk");
-            chunk.set_height_included(genesis_height);
-            chunk
-        })
-        .collect()
+    let mut chunks = vec![];
+
+    let num = shard_ids.len();
+    assert_eq!(state_roots.len(), num);
```

Comment:
nit: Does `num` need to be a separate variable? `shard_ids.len()` is clearer IMO.